Components/RTShaderSystem/src/OgreShaderProgramManager.cpp

```diff
@@ -294,6 +293,12 @@ GpuProgramPtr ProgramManager::createGpuProgram(Program* shaderProgram,
             programName, ResourceGroupManager::INTERNAL_RESOURCE_GROUP_NAME);
 
     if(pGpuProgram) {
+        if (!cachePath.empty())
```

Comment:
I think abusing `cachePath` for extended logging is not right. What do you thing about `#if OGRE_DEBUG_MODE`

Reply:
I agree - abusing cachePath is not ideal.

A solution which is more consistent with other Ogre classes (and a better approach for a library) would be to add a Listner interface to RTShader::ProgramManager. Something like:
```
class CacheListener
{
virtual void notifyGpuProgram(Program* program, Pass* pass,  const String& programName, const String& cachePath) = 0;
};
```